internal/cache/metadata/stat_cache.go

```diff
@@ -68,6 +68,15 @@ type StatCache interface {
 	// doesn't exist. Overwrite any existing entry for the name, positive or
 	// negative.
 	AddNegativeEntryForFolder(folderName string, expiration time.Time)
+
+	// Invalidate cache for all the entries with given prefix
+	// e.g. If cache contains below objects
+	// a/b
+	// a/d/c
+	// d
+	// Then it will invalidate entries a, a/b, a/d/c
```

Comment:
you didnt mention in the comment that cache has object a

Reply:
Added